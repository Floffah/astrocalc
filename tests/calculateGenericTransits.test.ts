import { describe, expect, test } from "bun:test";

import type { CalculateGenericTransitChartResponse } from "@/defs/responses.ts";
import app from "@/index.ts";

describe("Valid", () => {
    test("Call calculateGenericChart with valid parameters", async () => {
        const params = new URLSearchParams();

        params.set("year", "2000");
        params.set("month", "1");
        params.set("day", "1");
        params.set("hour", "5");
        params.set("minute", "30");

        const res = await app.request("/generic-chart?" + params.toString());

        const body = (await res.json()) as {
            data: CalculateGenericTransitChartResponse;
        };

        expect(body.data).toBeObject();
        expect(body.data.chart.signs).toMatchObject({
            moon: {
                cuspWarning: null,
                isVoidOfCourse: false,
                phase: "Waxing Crescent",
                value: "Scorpio",
            },
            sun: {
                cuspWarning: null,
                value: "Capricorn",
            },
        });

        expect(
            body.data.chart.planets.map(
                ({ degree: _, latitude: _1, longitude: _2, ...planet }) =>
                    planet,
            ),
        ).toMatchSnapshot();

        expect(
            body.data.chart.aspects.map(({ orb: _, ...aspect }) => aspect),
        ).toMatchSnapshot();

        expect(
            body.data.chart.declinations.map(
                ({ orb: _, ...declination }) => declination,
            ),
        ).toMatchSnapshot();

        expect(body.data.notableEvents).toMatchSnapshot();
    });
});
