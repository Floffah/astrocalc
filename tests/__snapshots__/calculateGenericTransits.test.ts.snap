// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Valid Call calculateGenericChart with valid parameters 1`] = `
[
  {
    "id": 0,
    "isRetrograde": false,
    "name": "Sun",
    "zodiac": {
      "id": 9,
      "lord": {
        "id": 6,
        "name": "Saturn",
      },
      "name": "Capricorn",
    },
  },
  {
    "id": 1,
    "isRetrograde": false,
    "name": "Moon",
    "zodiac": {
      "id": 7,
      "lord": {
        "id": 4,
        "name": "Mars",
      },
      "name": "Scorpio",
    },
  },
  {
    "id": 2,
    "isRetrograde": false,
    "name": "Mercury",
    "zodiac": {
      "id": 9,
      "lord": {
        "id": 6,
        "name": "Saturn",
      },
      "name": "Capricorn",
    },
  },
  {
    "id": 3,
    "isRetrograde": false,
    "name": "Venus",
    "zodiac": {
      "id": 8,
      "lord": {
        "id": 5,
        "name": "Jupiter",
      },
      "name": "Sagittarius",
    },
  },
  {
    "id": 4,
    "isRetrograde": false,
    "name": "Mars",
    "zodiac": {
      "id": 10,
      "lord": {
        "id": 6,
        "name": "Saturn",
      },
      "name": "Aquarius",
    },
  },
  {
    "id": 5,
    "isRetrograde": false,
    "name": "Jupiter",
    "zodiac": {
      "id": 0,
      "lord": {
        "id": 4,
        "name": "Mars",
      },
      "name": "Aries",
    },
  },
  {
    "id": 6,
    "isRetrograde": true,
    "name": "Saturn",
    "zodiac": {
      "id": 1,
      "lord": {
        "id": 2,
        "name": "Venus",
      },
      "name": "Taurus",
    },
  },
  {
    "id": 7,
    "isRetrograde": false,
    "name": "Uranus",
    "zodiac": {
      "id": 10,
      "lord": {
        "id": 6,
        "name": "Saturn",
      },
      "name": "Aquarius",
    },
  },
  {
    "id": 8,
    "isRetrograde": false,
    "name": "Neptune",
    "zodiac": {
      "id": 10,
      "lord": {
        "id": 6,
        "name": "Saturn",
      },
      "name": "Aquarius",
    },
  },
  {
    "id": 9,
    "isRetrograde": false,
    "name": "Pluto",
    "zodiac": {
      "id": 8,
      "lord": {
        "id": 5,
        "name": "Jupiter",
      },
      "name": "Sagittarius",
    },
  },
  {
    "id": 15,
    "isRetrograde": false,
    "name": "Chiron",
    "zodiac": {
      "id": 8,
      "lord": {
        "id": 5,
        "name": "Jupiter",
      },
      "name": "Sagittarius",
    },
  },
  {
    "id": 105,
    "isRetrograde": false,
    "name": "Lilith",
    "zodiac": {
      "id": 8,
      "lord": {
        "id": 5,
        "name": "Jupiter",
      },
      "name": "Sagittarius",
    },
  },
  {
    "id": 103,
    "isRetrograde": true,
    "name": "True North Node",
    "zodiac": {
      "id": 4,
      "lord": {
        "id": 0,
        "name": "Sun",
      },
      "name": "Leo",
    },
  },
  {
    "id": 104,
    "isRetrograde": true,
    "name": "True South Node",
    "zodiac": {
      "id": 4,
      "lord": {
        "id": 0,
        "name": "Sun",
      },
      "name": "Leo",
    },
  },
]
`;

exports[`Valid Call calculateGenericChart with valid parameters 2`] = `
[
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 1,
      "name": "Moon",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 0,
      "name": "Conjunction",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 2,
      "name": "Mercury",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 3,
      "name": "Semi Square",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 4,
      "name": "Mars",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 6,
      "name": "Saturn",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 9,
      "name": "Pluto",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 1,
      "name": "Opposition",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 6,
      "name": "Saturn",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 9,
      "name": "Pluto",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 3,
      "name": "Semi Square",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 1,
      "name": "Moon",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 3,
      "name": "Venus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 4,
      "name": "Mars",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 3,
      "name": "Semi Square",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 7,
      "name": "Semi Sextile",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 0,
      "name": "Conjunction",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 8,
      "name": "Quincunx",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 8,
      "name": "Quincunx",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 4,
      "name": "Mars",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 10,
      "name": "Bi Quintile",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 9,
      "name": "Quintile",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 9,
      "name": "Quintile",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 6,
      "name": "Saturn",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 9,
      "name": "Quintile",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 4,
      "name": "Sesquiquadrate",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 9,
      "name": "Pluto",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 4,
      "name": "Sesquiquadrate",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 8,
      "name": "Quincunx",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 9,
      "name": "Pluto",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 8,
      "name": "Quincunx",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 4,
      "name": "Sesquiquadrate",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 2,
      "name": "Square",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 7,
      "name": "Uranus",
    },
    "planet2": {
      "id": 9,
      "name": "Pluto",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 6,
      "name": "Sextile",
    },
    "planet1": {
      "id": 7,
      "name": "Uranus",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 1,
      "name": "Opposition",
    },
    "planet1": {
      "id": 8,
      "name": "Neptune",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 1,
      "name": "Opposition",
    },
    "planet1": {
      "id": 8,
      "name": "Neptune",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 0,
      "name": "Conjunction",
    },
    "planet1": {
      "id": 9,
      "name": "Pluto",
    },
    "planet2": {
      "id": 15,
      "name": "Chiron",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 9,
      "name": "Pluto",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 9,
      "name": "Pluto",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 15,
      "name": "Chiron",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 5,
      "name": "Trine",
    },
    "planet1": {
      "id": 15,
      "name": "Chiron",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
  {
    "aspect": {
      "id": 0,
      "name": "Conjunction",
    },
    "planet1": {
      "id": 103,
      "name": "True North Node",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "transit-to-transit",
  },
]
`;

exports[`Valid Call calculateGenericChart with valid parameters 3`] = `
[
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 2,
      "name": "Mercury",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 0,
      "name": "Sun",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 4,
      "name": "Mars",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 12,
      "name": "Contraparallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 2,
      "name": "Mercury",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 12,
      "name": "Contraparallel",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 12,
      "name": "Contraparallel",
    },
    "planet1": {
      "id": 3,
      "name": "Venus",
    },
    "planet2": {
      "id": 6,
      "name": "Saturn",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 5,
      "name": "Jupiter",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 12,
      "name": "Contraparallel",
    },
    "planet1": {
      "id": 4,
      "name": "Mars",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 5,
      "name": "Jupiter",
    },
    "planet2": {
      "id": 7,
      "name": "Uranus",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 12,
      "name": "Contraparallel",
    },
    "planet1": {
      "id": 6,
      "name": "Saturn",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 7,
      "name": "Uranus",
    },
    "planet2": {
      "id": 8,
      "name": "Neptune",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 7,
      "name": "Uranus",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 7,
      "name": "Uranus",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 8,
      "name": "Neptune",
    },
    "planet2": {
      "id": 103,
      "name": "True North Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 8,
      "name": "Neptune",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 15,
      "name": "Chiron",
    },
    "planet2": {
      "id": 105,
      "name": "Lilith",
    },
    "typeOfAspect": "natal-to-natal",
  },
  {
    "aspect": {
      "id": 11,
      "name": "Parallel",
    },
    "planet1": {
      "id": 103,
      "name": "True North Node",
    },
    "planet2": {
      "id": 104,
      "name": "True South Node",
    },
    "typeOfAspect": "natal-to-natal",
  },
]
`;

exports[`Valid Call calculateGenericChart with valid parameters 4`] = `
{
  "ingresses": [
    {
      "enteredSign": "Scorpio",
      "planet": "Moon",
    },
    {
      "enteredSign": "Capricorn",
      "planet": "Mercury",
    },
  ],
  "retrogradePlanets": [
    "Saturn",
    "True North Node",
    "True South Node",
  ],
}
`;
